cub/cub/device/device_reduce.cuh

```diff
@@ -451,11 +437,47 @@ public:
                   "Determinism should be used inside requires to have an effect.");
     using requirements_t =
       _CUDA_STD_EXEC::__query_result_or_t<EnvT, _CUDA_EXEC::__get_requirements_t, _CUDA_STD_EXEC::env<>>;
-    using determinism_t =
+    using default_determinism_t =
       _CUDA_STD_EXEC::__query_result_or_t<requirements_t, //
                                           _CUDA_EXEC::determinism::__get_determinism_t,
                                           _CUDA_EXEC::determinism::run_to_run_t>;
 
+    using accum_t = ::cuda::std::__accumulator_t<ReductionOpT, detail::it_value_t<InputIteratorT>, T>;
+
+    constexpr auto gpu_gpu_determinism =
+      ::cuda::std::is_same_v<default_determinism_t, ::cuda::execution::determinism::gpu_to_gpu_t>;
+
+    // integral types are always gpu-to-gpu deterministic, so fallback to run-to-run determinism
+    constexpr auto integral_fallback = gpu_gpu_determinism && ::cuda::std::is_integral_v<accum_t>;
+
+    // any floating point type with ::cuda::minimum<> or ::cuda::maximum<> are always gpu-to-gpu deterministic, so
+    // fallback to run-to-run determinism
+    constexpr auto fp_min_max_fallback =
+      gpu_gpu_determinism
+      && (::cuda::is_floating_point_v<accum_t>
+          && detail::is_one_of_v<ReductionOpT, ::cuda::minimum<>, ::cuda::maximum<>>);
```

Comment:
Question: If I am reducing an array of `float`s, using `cuda::minimum<>` is ok, but `cuda::minimum<float>` is not. Is this on purpose?